lib/Sema/TypeCheckDeclPrimary.cpp

```diff
@@ -627,7 +627,8 @@ CheckRedeclarationRequest::evaluate(Evaluator &eval, ValueDecl *current) const {
     // Thwart attempts to override the same declaration more than once.
     const auto *currentOverride = current->getOverriddenDecl();
     const auto *otherOverride = other->getOverriddenDecl();
-    if (currentOverride && currentOverride == otherOverride) {
+    const auto *otherInit = dyn_cast<ConstructorDecl>(other);
+    if (currentOverride && currentOverride == otherOverride && !(otherInit && otherInit->isImplicit())) {
```

Comment:
One nit: This doesn't look like it has been formatted by clang-format. I'd suggest you use utility from llvm-project repo - `llvm-project/clang/tools/clang-format`, you just need to add that to your PATH and run it after `git add`. I think some IDEs also support that as a formatter...

Reply:
Done